### PR TITLE
Button - The `aria-label` attribute is not changed by specifying it in the elementAttr option (T1115877)

### PR DESCRIPTION
**Pull request by EugeniyKiyashko**

### CODE REVIEW

**Review by Zedwag — DISMISSED** [review]

**Review by marker-dao — DISMISSED** [review]

File: testing/tests/DevExpress.ui.widgets/button.markup.tests.js [hunk 1]
```diff
@@ -260,6 +260,29 @@ QUnit.module('Button markup', function() {
             assert.equal($element.attr('aria-label'), undefined, 'aria label without text and icon is correct');
         });
 
+        QUnit.test('aria-label attribute should be override by custom value via elementAttr option', function(assert) {
```
Comment (ksercs) [inline testing/tests/DevExpress.ui.widgets/button.markup.tests.js]:
```suggestion
        QUnit.test('aria-label attribute should be overriden by custom value via elementAttr option (T1115877)', function(assert) {
```

File: js/renovation/ui/common/__tests__/widget.test.tsx [hunk 2]
```diff
@@ -917,6 +917,15 @@ describe('Widget', () => {
           });
         });
 
+        it('custom ARIA attributes should be more priority than aria property', () => {
+          const widget = new Widget({ visible: false, aria: { label: 'default' } });
+          widget.restAttributes = { 'aria-label': 'custom', 'aria-hidden': 'false' };
+
+          expect(widget.attributes).toEqual({
+            'aria-hidden': 'false', 'aria-label': 'custom',
```
Comment (ksercs) [inline js/renovation/ui/common/__tests__/widget.test.tsx]: It's more readable to me when we write objects this way. If you agree, let's refactor all objects in this test

```suggestion
            'aria-hidden': 'false', 
            'aria-label': 'custom',
```

File: js/renovation/ui/common/__tests__/widget.test.tsx [hunk 3]
```diff
@@ -917,6 +917,15 @@ describe('Widget', () => {
           });
         });
 
+        it('custom ARIA attributes should be more priority than aria property', () => {
```
Comment (ksercs) [inline js/renovation/ui/common/__tests__/widget.test.tsx]:
```suggestion
        it('rest attributes should have more priority than aria property (T1115877)', () => {
```

**Comment by ksercs** [comment]
@MikeVitik  there are changes in widget.tsx, review it plz

File: js/renovation/ui/common/widget.tsx [hunk 4]
```diff
@@ -420,9 +420,11 @@ export class Widget extends JSXComponent(WidgetProps) {
     } = this.props;
 
     const accessKey = focusStateEnabled && !disabled && this.props.accessKey;
+
     return {
-      ...extend({}, this.restAttributes, accessKey && { accessKey }) as Record<string, string>,
+      ...extend({}, accessKey && { accessKey }) as Record<string, string>,
       ...getAria({ ...aria, disabled, hidden: !visible }),
+      ...this.restAttributes,
```
Comment (ksercs) [inline js/renovation/ui/common/widget.tsx]: seems like we should cast it to `Record<string, stirng>`.
I'm also not sure that `restAttributes` can not be `undefined` for example. Maybe it's better to write this way:
`...extend({}, this.restAttributes)  as Record<string, string>`

Comment (EugeniyKiyashko) [reply js/renovation/ui/common/widget.tsx]: good remark. I am not sure too https://github.com/DevExpress/DevExtreme/pull/22673/commits/deeb89db3457cca422513881aeded6e459ca0ad3

**Review by MikeVitik — DISMISSED** [review]

File: js/renovation/ui/common/__tests__/widget.test.tsx [hunk 5]
```diff
@@ -917,6 +917,16 @@ describe('Widget', () => {
           });
         });
 
+        it('rest attributes should have more priority than aria property (T1115877)', () => {
+          const widget = new Widget({ visible: false, aria: { label: 'default' } });
+          widget.restAttributes = { 'aria-label': 'custom', 'aria-hidden': 'false' };
+
+          expect(widget.attributes).toEqual({
+            'aria-hidden': 'false',
```
Comment (ksercs) [inline js/renovation/ui/common/__tests__/widget.test.tsx]: there is a trailing space :(

**Review by Zedwag — APPROVED** [review]

**Pull request merged**